scripts/api-contract/response_shapes.py

```
@@ -0,0 +1,133 @@
+    return [
+        f"{'/'.join(map(str, e.path)) or '(root)'}: {e.message}"
+        for e in sorted(validator.iter_errors(body), key=lambda e: list(e.path))
+    ]
+
+
+def fetch(base, path, token, workspace):
+    sep = "&" if "?" in path else "?"
+    req = urllib.request.Request(
+        f"{base}{path}{sep}workspace_id={workspace}",
+        headers={"Authorization": f"Bearer {token}", "X-Workspace-ID": workspace},
+    )
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)